GTBitsOfGood/southface
# Nabeel/80 report UI

## Report Builder Functionality and UI

Issue Number(s): #2 

This PR introduces functionality for building an active report using SWR functions across several places in the application. It touches many parts of the codebase and introduces major changes to the database.

### What works:

- Active report is created upon clicking "Add to report"
- Images can be added to and removed from report
- Notes can be added to and removed from report
- Report builder page matches design (besides completed reports view)
- Report builder page has active report editing functionalities
- Library page has active report editing functionality
- Recent standards view has active report editing functionality

### What doesn't work:
- Tested only on local db; possible conflicts may occur on shared DB
- Renaming, saving active report doesn't work
- completed reports view is incomplete
- shopping cart view is incomplete
- login/logout conditions, error handling

### How to Test
- login as admin
- Add cards to report
- Remove cards from report
- Go into a modal and try selecting/deselecting images
- Go to report builder and do the same things
- Try modifying report from recent standards view
- Verify edits are reflected in database
- Verify edits behave as expected

Dude this was so much work smh
